### DISCUSSION TITLE
Do APN and PrinterID need to be unique across devices on a print farm?

### DISCUSSION THREAD

**Discussion by jrytio** [gdombiak/OctoPod#493] [Q&A]
@gdombiak I'm building out an automated deployment of my OctoPrint farm using Balena.io.   To do so, I'm storing the config.yaml in Git and then changing some of the machine specific parameters at run time.  

I've deployed this to 3 test printers and it all seems to be working when I duplicate the APN and PrinterID, though if I leave the printer name the same, all alerts to my watch will say 'Dev3' or whatever name I have in source.   So I'm changing that dynamically for each machine as well.

It works fine on my side to just change the name and leave APN and PrinterID the same across devices.   I just wanted to check that I'm not causing any undue harm to the backend of your notifications by doing this?

**Comment by gdombiak** [comment]
Hi @tideline3d,

Cool idea and project! I like it. Backend is safe no matter what setup you have. Being an open source project where I pay hosting of the server for push notifications ... the only considering is volume of traffic. So far we are within the limits so we should be good.

As long as your setup sends the push notification to the proper phone about the proper printer ... you are doing good!

Regards,
Gaston

**Reply by jrytio** [reply]
Perfect, I ended up getting it working on all 9 printers.  Now I can just push Octopod plug-in updates with a small change to a file and it'll deploy to the whole farm, waiting until they finished printing if they're mid-print. 

Love your work, use it every day. Thanks again!

**Reply by gdombiak** [reply]
That's great to hear @tideline3d! And thanks for the kind words. Happy to hear that you found value using OctoPod. 

BTW, big thanks for becoming a sponsor. I appreciate your help! Let me know if you have any feature that you would like to see added. Try to give higher priority to sponsors. :)

Take care,
Gaston
